torch/special/__init__.py

```diff
@@ -159,6 +159,37 @@
     tensor([-0.9466,  2.6352,  0.6131, -1.7169,  0.6261])
 """.format(**common_args))
 
+logsumexp = _add_docstr(_special.special_logsumexp,
+                        r"""
+logsumexp(input, dim, keepdim=False, *, out=None)
+
+Returns the log of summed exponentials of each row of the :attr:`input`
+tensor in the given dimension :attr:`dim`. The computation is numerically
+stabilized.
```

Comment:
We can take this chance to improve the wording of the documentation. What about just saying:

> Computes :math:\`\log \sum \exp (x)\` for an input `x` along the dimension :attr:\`dim\` with a numerically stable algorithm.

The math formula could be replaced even by `input.exp().sum(dim).log()`. What do you think @mruberry ?

Reply:
I agree that this is better wording to the documentation, but do you think that this introduces redundancy (for formula) in the doc? As the formula is already mentioned just below this:

```
For summation index :math:`j` given by `dim` and other indices :math:`i`, the result is

    .. math::
        \text{{logsumexp}}(x)_{{i}} = \log \sum_j \exp(x_{{ij}})
```

I think for those who may not want to read the formula first, having it explained in words is OK. I maybe wrong but just sharing my thoughts. Looking forward to hear your opinion on this, @Lezcano @mruberry!

Reply:
Let's not sweat the documentation in this pr